src/common/Session.cpp

```diff
@@ -133,26 +133,34 @@ namespace SDDM {
         m_valid = false;
         m_desktopNames.clear();
 
+        QStringList sessionDirs;
+
         switch (type) {
         case WaylandSession:
-            m_dir = QDir(mainConfig.Wayland.SessionDir.get());
+            sessionDirs = mainConfig.Wayland.SessionDir.get();
             m_xdgSessionType = QStringLiteral("wayland");
             break;
         case X11Session:
-            m_dir = QDir(mainConfig.X11.SessionDir.get());
+            sessionDirs = mainConfig.X11.SessionDir.get();
             m_xdgSessionType = QStringLiteral("x11");
             break;
         default:
             m_xdgSessionType.clear();
             break;
         }
 
-        m_fileName = m_dir.absoluteFilePath(fileName);
+        QFile file;
+        for (const QDir dir: qAsConst(sessionDirs)) {
```

Comment:
IMO it should be more explicit that this creates a `QDir` from a path. It could also check that the dir exists.

Reply:
Done.
For the check, it didn't exist before and should work fine without, so I would prefer we add it in a separate commit if we need it for whatever reason.